pkg/cmd/server/crypto/crypto.go

```diff
@@ -335,10 +337,10 @@ func MakeCA(certFile, keyFile, serialFile, name string) (*CA, error) {
 	}, nil
 }
 
-func (ca *CA) EnsureServerCert(certFile, keyFile string, hostnames sets.String) (*TLSCertificateConfig, bool, error) {
+func (ca *CA) EnsureServerCert(certFile, keyFile string, hostnames sets.String, expirationDate time.Time) (*TLSCertificateConfig, bool, error) {
```

Comment:
> i would rather make it part of CA struct

After closer look at it I think it's better to leave it as-is, because CA stands for "Certification Authority" and expiration date is a property of the certificate itself rather than CA.